### PR TITLE
first version 

### PR DESCRIPTION
**Pull request by ChamseddineBhd**
Signed-off-by: chamseddine <chamseddine.benhamed@ensi-uma.tn>

This PR should be merged before: https://github.com/gridsuite/study-server/pull/92

### CODE REVIEW

File: src/translations/fr.json [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
-    "paramsRetrievingError": "Une erreur est survenue lors de la récupération des paramètres"
+    "paramsRetrievingError": "Une erreur est survenue lors de la récupération des paramètres",
+    "elementUuid": "Uuid de l'élement",
+    "elementName": "Nom",
+    "owner": "Propriétaire",
+    "type": "Type",
+    "private": "privé",
+    "public": "publique",
+    "emptyDir": "Dossier vide",
+    "accessRights": "Droits d'accés"
```
Comment (FranckLecuyer) [inline src/translations/fr.json]: accès

File: src/translations/fr.json [hunk 2]
```diff
@@ -1,5 +1,13 @@
 {
     "parameters": "Paramètres",
     "paramsChangingError": "Une erreur est survenue lors de la modification des paramètres",
-    "paramsRetrievingError": "Une erreur est survenue lors de la récupération des paramètres"
+    "paramsRetrievingError": "Une erreur est survenue lors de la récupération des paramètres",
+    "elementUuid": "Uuid de l'élement",
```
Comment (FranckLecuyer) [inline src/translations/fr.json]: élément

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/components/custom-tree-view.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import { useDispatch, useSelector } from 'react-redux';
+import { setCurrentChildren, setSelectedDirectory } from '../redux/actions';
+
+const useStyles = makeStyles((theme) => ({
+    paper: {
+        height: 140,
+        width: 100,
+    },
+    control: {
+        padding: theme.spacing(2),
```
Comment (niconoir) [inline src/components/custom-tree-view.js]: Styles "paper" and "control" are defined but never used further in the code. Is it normal ?

File: src/utils/rest-api.js [hunk 4]
```diff
@@ -96,6 +98,29 @@ export function fetchConfigParameter(name) {
     );
 }
 
+export function fetchDirectoryContent(directoryUuid) {
+    console.info("Fetching Folder content '%s'", directoryUuid);
+    const fetchParams =
```
Comment (niconoir) [inline src/utils/rest-api.js]: Change variable name to something more accurate than fetchParams

File: src/utils/rest-api.js [hunk 5]
```diff
@@ -96,6 +98,29 @@ export function fetchConfigParameter(name) {
...
+    return backendFetch(fetchParams).then((response) =>
+        response.ok
+            ? response.json()
+            : response.text().then((text) => Promise.reject(text))
+    );
+}
+
+export function fetchRootFolders() {
+    console.info('Fetching Root Directories');
+    const fetchParams =
```
Comment (niconoir) [inline src/utils/rest-api.js]: Change variable name to something more accurate than fetchParams

File: src/utils/rest-api.js [hunk 6]
```diff
@@ -114,3 +139,18 @@ export function updateConfigParameter(name, value) {
             : response.text().then((text) => Promise.reject(text))
     );
 }
+
+export function fetchStudiesInfos(uuids) {
+    console.info('Fetching studies metadata ... ');
+    const fetchParams = PREFIX_STUDY_SERVER_QUERIES + `/v1/studies/metadata`;
```
Comment (niconoir) [inline src/utils/rest-api.js]: Change variable name to something more accurate than fetchParams

File: src/components/directory-content.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+import Tooltip from '@material-ui/core/Tooltip';
+import { fetchStudiesInfos } from '../utils/rest-api';
+
+const useStyles = makeStyles((theme) => ({
+    link: {
+        color: theme.link.color,
+        textDecoration: 'none',
+    },
+    tablePaper: {
```
Comment (niconoir) [inline src/components/directory-content.js]: tablePaper style is not used further in the code

File: src/translations/fr.json [hunk 8]
```diff
@@ -1,5 +1,13 @@
...
+    "elementName": "Nom",
+    "owner": "Propriétaire",
+    "type": "Type",
+    "private": "privé",
+    "public": "publique",
```
Comment (niconoir) [inline src/translations/fr.json]: I would have translated it as 'public'

**Review by niconoir — COMMENTED** [review]
The virtualized table component is almost the same as the one used in gridstudy-app. Maybe it could be mutualized in commons-ui ?

File: src/translations/fr.json (hunk 1 above)
Comment (FranckLecuyer) [reply src/translations/fr.json]: 'accès' instead of  'accés'

**Review by FranckLecuyer — APPROVED** [review]
Corrections done

File: src/components/custom-tree-view.js [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    },
+}));
+
+const CustomTreeView = ({ rootDirectory }) => {
```
Comment (geofjamg) [inline src/components/custom-tree-view.js]: You should give it a more business name, like DirectoryTreeView

File: src/components/util/virtualized-table.js [hunk 10]
```diff
@@ -0,0 +1,381 @@
+/**
+ * This class has been taken from 'Virtualized Table' example at https://material-ui.com/components/tables/
```
Comment (geofjamg) [inline src/components/util/virtualized-table.js]: Is it a copy paste from study? Maybe this is the rigth time to share it in commons ?

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: src/translations/fr.json (hunk 8 above)
Comment (geofjamg) [reply src/translations/fr.json]: Yes! Look at public vs publique

File: src/redux/reducer.js [hunk 11]
```diff
@@ -31,8 +34,11 @@ const paramsInitialState = {
 
 const initialState = {
     computedLanguage: getLocalStorageComputedLanguage(),
+    currentChildren: null,
```
Comment (geofjamg) [inline src/redux/reducer.js:37]: What is the difference between  currentChildren and selectedDirectory ?

Comment (ChamseddineBhd) [reply src/redux/reducer.js:37]: currentChildren are the children of a selectedDirectory (studies objects etc...)

File: src/components/util/virtualized-table.js (hunk 10 above)
Comment (ChamseddineBhd) [reply src/components/util/virtualized-table.js]: i'll move it in a separate PR in gridexplore and gridstudy

Comment (geofjamg) [reply src/components/util/virtualized-table.js]: No, it has to be done before this PR

**Pull request merged**